index.js

```diff
@@ -546,15 +517,13 @@ class KeyringController extends EventEmitter {
    * @param {string} password - The keyring controller password.
    * @returns {Promise<boolean>} Resolves to true once keyrings are persisted.
    */
-  persistAllKeyrings(password = this.password) {
+  async persistAllKeyrings(password = this.password) {
     if (typeof password !== 'string') {
-      return Promise.reject(
-        new Error('KeyringController - password is not a string'),
-      );
+      throw new Error('KeyringController - password is not a string');
     }
 
     this.password = password;
-    return Promise.all(
+    const serializedKeyrings = await Promise.all(
       this.keyrings.map((keyring) => {
         return Promise.all([keyring.type, keyring.serialize()]).then(
```

Comment:
Nit: We can replace the `.then` here with an `await` as well

Reply:
Done!